client/state/states-list/schema.js

```diff
@@ -0,0 +1,17 @@
+export const statesListSchema = {
```

Comment:
I think the following should work. When can also test to see if a schema works with unit tests. See example [here](https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/media-storage/test/reducer.js#L123-L169)

We can also use http://jsonschema.net/#/ as a good starting point to see what the schema structure might look like.
```javascript
export const statesListSchema = {
  type: 'object',
  //otherwise we'll accept extra props that don't match the regex
  additionalProperties: false,
  patternProperties: {
    '^[A-Z]{2}$': {
      type: 'array',
      items: {
        type: 'object',
        properties: {
          code: {
            type: 'string'
          },
          name: {
            type: 'string'
          }
        },
        required: [
          'code',
          'name'
        ]
      }
    }
  }
};
```